src/test/java/org/apache/commons/jxpath/ri/StressTest.java

```diff
@@ -47,25 +46,16 @@ public void testThreads() throws Throwable {
         }
 
         for (final Thread element : threadArray) {
-            try {
-                element.join();
-            }
-            catch (final InterruptedException e) {
-                fail("Interrupted");
-            }
-        }
-
-        if (exception != null) {
-            throw exception;
+            assertDoesNotThrow(() -> element.join(), "Interrupted");
```

Comment:
Same comment as before RE `assertDoesNotThrow()`. You'd get the same information from the failure if the exception percolated to JUnit.